src/main/java/com/jfrog/ide/idea/scan/ScanBinaryExecutor.java

```diff
@@ -288,7 +287,7 @@ protected void downloadBinary() throws IOException {
         }
     }
 
-    Path createTempRunInputFile(ScansConfig scanInput) throws IOException {
+    Path createTempRunInputFile(Object scanInput) throws IOException {
```

Comment:
Deserializing an object is considered insecure. See https://owasp.org/www-project-top-ten/2017/A8_2017-Insecure_Deserialization.
Let's deserialize only ScanConfig or NewScanConfig. You can use method overloading for this.

Reply:
Those are internal input objects that should be unified in the near future (when all the scanners will support the new config file). Hence we are only deserializing objects we are creating, I see no security concern here.